### PR TITLE
Refactor: CLI binaries integration

### LINKED ISSUE #10: Support new sentry-cli architectures
Similar to https://github.com/getsentry/sentry-android-gradle-plugin/issues/192

https://github.com/getsentry/sentry-dart-plugin/blob/main/lib/src/configuration.dart#L146-L169

https://github.com/getsentry/sentry-dart-plugin/tree/main/lib/assets

### PR DESCRIPTION
**Pull request by vaind**
_#skip-changelog_

## :scroll: Description

to enable automatic updates in CI, as seen in https://github.com/getsentry/sentry-unity/pull/855

as a side-effect, closes #10

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] I updated the docs if needed
- [ ] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

* CI for CLI update PRs
* packaging in CI so that the person who does publishing to pub.dev doesn't forget to download the CLI?

### CODE REVIEW

File: lib/src/configuration.dart [hunk 1]
```diff
@@ -149,7 +149,7 @@ class Configuration {
 
   void _findAndSetCliPath() {
     if (Platform.isMacOS) {
-      _setCliPath("Darwin-x86_64");
+      _setCliPath("Darwin-universal");
```
Comment (marandaneto) [inline lib/src/configuration.dart:152]: That requires updating https://github.com/getsentry/sentry-dart-plugin/blob/cc136ca78825d28dc08c2a231ea6b7129d1215b0/lib/src/configuration.dart#L150-L162
And with that, we could close https://github.com/getsentry/sentry-dart-plugin/issues/10

File: scripts/sentry-cli.properties [hunk 2]
```diff
@@ -0,0 +1,2 @@
+version = 1.69.1
```
Comment (marandaneto) [inline scripts/sentry-cli.properties:1]: We can likely bump to https://github.com/getsentry/sentry-cli/releases/tag/1.74.4

Comment (vaind) [reply scripts/sentry-cli.properties:1]: wanted to let auto-update CI do that :)

Comment (vaind) [reply scripts/sentry-cli.properties:1]: but wait, there're no good tests in this repo... so bumping to v2.0 should be done with manual testing...

OK, updating to 1.74.4 now
```suggestion
version = 1.74.4
```

File: lib/src/configuration.dart (hunk 1 above)
Comment (marandaneto) [reply lib/src/configuration.dart:152]: Ops, missed that, thanks.

**Comment by marandaneto** [comment]
@bruno-garcia please give permission to @vaind so we don't need to approve CI every time.

**Review by philipphofmann — COMMENTED** [review]
The path of the `download-sentry-cli.sh` changed, but I see no changes related to that somewhere in this PR. Am I missing something? Who is calling this?

**Review by marandaneto — APPROVED** [review]

**Pull request merged**